transport/grpc/handler_test.go

```diff
@@ -64,3 +67,32 @@ func TestResponseWriter_Write(t *testing.T) {
 	assert.Equal(t, error(nil), err)
 	assert.Equal(t, strMsg, string(r.body.Bytes()))
 }
+
+func TestResponseWriter_AddHeaders(t *testing.T) {
+	caller := "teeeeest"
+	encoding := "raw"
+	inputHeaders := transport.HeadersFromMap(map[string]string{
+		CallerHeader:   caller,
+		EncodingHeader: encoding,
+	})
+	expectedHeaders := metadata.New(map[string]string{
+		ApplicationHeaderPrefix + CallerHeader:   caller,
+		ApplicationHeaderPrefix + EncodingHeader: encoding,
+	})
+
+	var r response
+	rw := newResponseWriter(&r)
+
+	rw.AddHeaders(inputHeaders)
+
+	assert.Equal(t, expectedHeaders, r.headers)
+}
+
+func TestResponseWriter_SetApplicationError(t *testing.T) {
+	var r response
+	rw := newResponseWriter(&r)
+
+	rw.SetApplicationError()
+
+	// No action on Application Error
```

Comment:
If there's no behavioral change, I wouldn't even bother with testing it. If
you just want to make sure that that code path is exercised, crossdock will
take care of it.